crates/tx-parser/src/filter.rs

```diff
@@ -24,6 +25,77 @@ pub enum TxFilterRule {
     DepositRequest(DepositTxParams),
     /// Deposit transaction with deposit config and address
     Deposit(DepositTxParams),
+    /// Addresses that are spent to
+    SpentToAddrs(Vec<BitcoinAddress>),
+    /// Blob ids that are expected
+    BlobIds(Vec<Buf32>),
```

Comment:
These vecs can be sorted and then we can do binary search.
Having `Vec` instead of Individual items lets us group the items, sort and do binary search as we want.

Reply:
Yes, as I think about it more now, having a separate rule with enum variant looks kind of redundant. Basically I want a particular type/variant to produce a specific ProtocolOp,k but yes, that can be done with structs that wrap some value without the need of these enums.